File: backend/internal/models/concert_category.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type ConcertCategory struct {
	// gorm.Model
	ID               uuid.UUID `gorm:"unique;type:uuid;primaryKey"`
	ConcertId        uuid.UUID `gorm:"type:uuid;not null;index"`
	CategoryId       int       `gorm:"not null;index"`
	AvailableTickets int       `gorm:"not null"`
	SoldTickets      int
	Price            float64 `gorm:"not null"`
	Tickets          []Ticket
	// Tickets          []Ticket `gorm:"-"`
	Concert   Concert  `gorm:"foreignKey:ConcertId"`
	Category  Category `gorm:"foreignKey:CategoryId"`
	CreatedAt time.Time
	UpdatedAt time.Time
	DeletedAt *time.Time `gorm:"index"`
}

File: frontend/lib/core/models/concert_category.dart
import 'package:weezemaster/core/models/category.dart';
import 'package:weezemaster/core/models/ticket.dart';

class ConcertCategory {
  final String id;
  final int availableTickets;
  final int soldTickets;
  final double price;
  final List<Ticket> tickets;
  final Category category;

  ConcertCategory({
    required this.id,
    required this.availableTickets,
    required this.soldTickets,
    required this.price,
    required this.tickets,
    required this.category 
  });

  factory ConcertCategory.fromJson(Map<String, dynamic> json) {
    var ticketListFromJson = json['Tickets'] as List? ?? [];
    List<Ticket> ticketList = ticketListFromJson.map((i) => Ticket.fromJson(i)).toList();

    return ConcertCategory(
      id: json['ID'],
      availableTickets: json['AvailableTickets'] ?? 0,
      soldTickets: json['SoldTickets'] ?? 0,
      price: (json['Price'] ?? 0).toDouble(),
      tickets: ticketList,
      category: Category.fromJson(json['Category']),
    );
  }
}